### PR TITLE
K8SPG-519: add `.spec.extensions.storage.endpoint`

### PR DESCRIPTION
**Pull request by pooknull**
[K8SPG-519](https://jira.percona.com/browse/K8SPG-519) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://perconadev.atlassian.net/browse/K8SPG-519

**DESCRIPTION**
---
**Problem:**
When installing custom extensions it is not possible to point to a custom s3 object storage endpoint.

**Cause:**
The s3 SDK within the extension installer is only minimally configured, currently using default values. 

**Solution:**
Add an optional flag to the extension installer named `endpoint` and configure the s3 SDK appropriately.

It's an improved version of the https://github.com/percona/percona-postgresql-operator/pull/672 PR. Since the branch https://github.com/timorusky/percona-postgresql-operator/tree/extensions-endpoint was archived, it was decided to continue development of the feature in a separate PR.

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?
- [x] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [x] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [ ] Is an E2E test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?

**Config/Logging/Testability**
- [x] Are all needed new/changed options added to default YAML files?
- [x] Are the manifests (crd/bundle) regenerated if needed?
- [x] Did we add proper logging messages for operator actions?
- [x] Did we ensure compatibility with the previous version or cluster upgrade process?
- [x] Does the change support oldest and newest supported PG version?
- [x] Does the change support oldest and newest supported Kubernetes version?

[K8SPG-519]: https://perconadev.atlassian.net/browse/K8SPG-519?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| custom-extensions | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/4bac4c128/custom-extensions.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/4bac4c128/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/4bac4c128/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/4bac4c128/monitoring.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/4bac4c128/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/4bac4c128/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/4bac4c128/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/4bac4c128/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/4bac4c128/start-from-backup.log) |
| tablespaces | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/4bac4c128/tablespaces.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/4bac4c128/telemetry-transfer.log) |
| upgrade-minor | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/4bac4c128/upgrade-minor.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/4bac4c128/users.log) |
| We run 13 out of 13|

commit: https://github.com/percona/percona-postgresql-operator/pull/787/commits/4bac4c128fcf20e26508fcda342a28e74101c85b
image: `perconalab/percona-postgresql-operator:PR-787-4bac4c128`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| custom-extensions | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/6acfd441c/custom-extensions.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/6acfd441c/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/6acfd441c/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/6acfd441c/monitoring.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/6acfd441c/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/6acfd441c/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/6acfd441c/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/6acfd441c/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/6acfd441c/start-from-backup.log) |
| tablespaces | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/6acfd441c/tablespaces.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/6acfd441c/telemetry-transfer.log) |
| upgrade-minor | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/6acfd441c/upgrade-minor.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/6acfd441c/users.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/6acfd441c/pitr.log) |
| We run 14 out of 14|

commit: https://github.com/percona/percona-postgresql-operator/pull/787/commits/6acfd441c6f656f38668a9ed33a989f3ce82a1f9
image: `perconalab/percona-postgresql-operator:PR-787-6acfd441c`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| custom-extensions | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/custom-extensions.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/monitoring.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/one-pod.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/operator-self-healing.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/pitr.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/start-from-backup.log) |
| tablespaces | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/tablespaces.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/telemetry-transfer.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/upgrade-consistency.log) |
| upgrade-minor | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/upgrade-minor.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/users.log) |
| We run 16 out of 16|

commit: https://github.com/percona/percona-postgresql-operator/pull/787/commits/7cabf7ff536a7d57a357c6dc7b319b3393a160b7
image: `perconalab/percona-postgresql-operator:PR-787-7cabf7ff5`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| custom-extensions | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/custom-extensions.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/monitoring.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/one-pod.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/operator-self-healing.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/pitr.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/start-from-backup.log) |
| tablespaces | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/tablespaces.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/telemetry-transfer.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/upgrade-consistency.log) |
| upgrade-minor | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/upgrade-minor.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/7cabf7ff5/users.log) |
| We run 16 out of 16|

commit: https://github.com/percona/percona-postgresql-operator/pull/787/commits/7cabf7ff536a7d57a357c6dc7b319b3393a160b7
image: `perconalab/percona-postgresql-operator:PR-787-7cabf7ff5`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| custom-extensions | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/616dcd2e7/custom-extensions.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/616dcd2e7/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/616dcd2e7/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/616dcd2e7/monitoring.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/616dcd2e7/one-pod.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/616dcd2e7/operator-self-healing.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/616dcd2e7/pitr.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/616dcd2e7/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/616dcd2e7/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/616dcd2e7/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/616dcd2e7/start-from-backup.log) |
| tablespaces | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/616dcd2e7/tablespaces.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/616dcd2e7/telemetry-transfer.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/616dcd2e7/upgrade-consistency.log) |
| upgrade-minor | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/616dcd2e7/upgrade-minor.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-787/616dcd2e7/users.log) |
| We run 16 out of 16|

commit: https://github.com/percona/percona-postgresql-operator/pull/787/commits/616dcd2e7995be078b904d42355a8546fc9629b8
image: `perconalab/percona-postgresql-operator:PR-787-616dcd2e7`

File: build/postgres-operator/Dockerfile [hunk 1]
```diff
@@ -19,7 +19,7 @@ RUN mkdir -p build/_output/bin \
        go build -ldflags "-w -s -X main.GitCommit=$GIT_COMMIT -X main.GitBranch=$GIT_BRANCH" \
            -o build/_output/bin/postgres-operator \
                ./cmd/postgres-operator \
-    && CGO_ENABLED=$CGO_ENABLED GOOS=$GOOS GO_LDFLAGS=$GO_LDFLAGS \
+    && CGO_ENABLED=$CGO_ENABLED GOARCH=${TARGETARCH} GOOS=$GOOS GO_LDFLAGS=$GO_LDFLAGS \
```
Comment (egegunes) [inline build/postgres-operator/Dockerfile:22]: why do we need it?

Comment (pooknull) [reply build/postgres-operator/Dockerfile:22]: I had a following error in `custom-extensions` test:
```
$ kubectl logs custom-extensions-instance1-g95j-0 extension-installer
+ IFS=,
+ read -ra extensions
+ args=('-type' 's3' '-region' 'eu-central-1' '-bucket' 'pg-extensions' '-extension-path' '/pgdata/extension/16')
+ declare -a args
+ [[ -n '' ]]
+ for key in "${extensions[@]}"
+ '[' -f /pgdata/extension/16/pg_cron-pg16-1.6.1.installed ']'
+ echo 'Installing extension: pg_cron-pg16-1.6.1'
Installing extension: pg_cron-pg16-1.6.1
+ /usr/local/bin/extension-installer -type s3 -region eu-central-1 -bucket pg-extensions -extension-path /pgdata/extension/16 -key pg_cron-pg16-1.6.1 -install
/usr/local/bin/install-extensions.sh: line 27: /usr/local/bin/extension-installer: cannot execute binary file: Exec format error
```
Specifying an architecture fixes it

**Review by hors — APPROVED** [review]

**Review by inelpandzic — APPROVED** [review]

**Review by egegunes — APPROVED** [review]

**Pull request merged**